src/main/java/com/patreon/euphrates/ScrubbingInputStream.java

```diff
@@ -6,90 +6,88 @@
 import java.io.FilterInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.CodingErrorAction;
 
 public class ScrubbingInputStream extends FilterInputStream {
 
-  private static final Logger LOG = LoggerFactory.getLogger(Replicator.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ScrubbingInputStream.class);
+  CharsetDecoder decoder = Charset.forName("UTF-8").newDecoder();
+  CharsetEncoder encoder = Charset.forName("UTF-8").newEncoder();
+  byte[] readBuffer = new byte[4096];
+  CharBuffer charBuffer, goodCharBuffer;
+  ByteBuffer byteBuffer;
 
   public ScrubbingInputStream(InputStream in) {
     super(in);
+    decoder.onMalformedInput(CodingErrorAction.IGNORE);
+    byteBuffer = ByteBuffer.allocate(4096);
+    charBuffer = CharBuffer.allocate(4096);
```

Comment:
I think this is twice as big as you need.  Once it's a Java char, it's UTF-16, and `allocate` is number of chars, not number of bytes.